textproc/opensearch: Update to 2.11.1

smortex opened the pull request:
The basic features I use work as expected.  Some of you use more advance features, so please report failures.

smortex committed:
textproc/opensearch: Update to 2.11.0

With hat:	opensearch

smortex committed:
textproc/opensearch-dashboards: Update to 2.11.0

With hat:	puppet

alonsobsd commented:
Hello, I was testing this version with wazuh-dashboard-plugins and it breaks completely (https://github.com/wazuh/wazuh-dashboard/issues/113). I guess it was because 2.11.0 removed angularjs. I think the best could be create a opensearch210-dashboard port for wazuh until it can works without problems with opensearch-dashboards >= 2.11.0.

smortex commented:
You mean this guy was right: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4993#issuecomment-1717986504 ? :unamused:

And people still ask [if it is acceptable to ship features in patch releases](https://github.com/opensearch-project/opensearch-build/issues/4161#issuecomment-1789764912) :roll_eyes: Good news, this is *not* what was accepted in the end.

This really gets boring…

+1 for a copy with the MAJOR + MINOR version in the port name, so `opensearch210-dashboard` or `opensearch-dashboard210` maybe more in-line with what is usually done in the ports tree.

hackacad commented:
Splitting the Port at a minor version does not feel right, but theses changes break certain plugins (maybe more than just Wazuh) we should definitely go for a *210 port. It's already too late for fixing/intervening upstream...

smortex committed:
textproc/opensearch-dashboards210: New port

OpenSeach-Dashboards 2.11.0 introduce some supposed internal changes
that break wazuh.  Before updating OpenSeach, make the old version
available with a new port.

With hat:	opensearch

smortex commented:
I inserted a commit to copy the opensearch-dashboards port. I think it make sense to do the same with opensearch itself, right?

smortex committed:
textproc/opensearch210: New port

OpenSeach-Dashboards 2.11.0 introduce some supposed internal changes
that break wazuh.  Before updating OpenSeach, make the old version
available with a new port.

With hat:	opensearch

hackacad commented:
> You mean this guy was right: [opensearch-project/OpenSearch-Dashboards#4993 (comment)](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4993#issuecomment-1717986504) ? 😒
> 
> And people still ask [if it is acceptable to ship features in patch releases](https://github.com/opensearch-project/opensearch-build/issues/4161#issuecomment-1789764912) 🙄 Good news, this is _not_ what was accepted in the end.
> 
> This really gets boring…
> 
> +1 for a copy with the MAJOR + MINOR version in the port name, so `opensearch210-dashboard` or `opensearch-dashboard210` maybe more in-line with what is usually done in the ports tree.

I see multiple new warnings after 2.10-2.11 upgrade:

```
org.opensearch.core.compress.NotXContentException: Compressor detection can only be called on some xcontent bytes or compressed xcontent bytes
        at org.opensearch.core.compress.CompressorRegistry.compressor(CompressorRegistry.java:75) ~[opensearch-core-2.11.0.jar:2.11.0]
```
        
IP Adress check has changed:

```
{"type":"log","@timestamp":"2023-11-13T15:45:32Z","tags":["error","opensearch","data"],"pid":5653,"message":"[ConnectionError]: connect ECONNREFUSED 127.0.0.1:9200"}
```

when using `server.host: 0.0.0.0` in a jail (this was just an upgrade, so behavior has definitely changed).

None if these looks like a major issue, but I'd like to do some more testing before pushing this commit.

smortex commented:
> [...]
> None if these looks like a major issue, but I'd like to do some more testing before pushing this commit.

Sure! I also experienced some breakage related to the listen address on Linux deployments, but did not dig in (in my case OS listening on localhost was only listening on `::1` while OSD was using `127.0.0.1` to reach OS).

I added commits to copy the existing ports with a 210 suffix. I have not tested it since I don't use wazuh.  Please report feedback if you give it a try!

alonsobsd committed:
Update Makefile

alonsobsd committed:
Update Makefile

alonsobsd committed:
Update Makefile

alonsobsd committed:
Update Makefile

alonsobsd committed:
Update Makefile

Use opensearch-dashboards like MASTER port

alonsobsd committed:
Update Makefile

Use textproc/opensearch-dashboards like MASTER port

alonsobsd committed:
Update Makefile

Use textproc/opensearch-dashboards like MASTERDIR

alonsobsd committed:
Update Makefile

Use opensearch-dashboards like MASTERDIR

alonsobsd committed:
Delete textproc/opensearch-dashboards210/files/opensearch-dashboards.in

Delete obsolete file

alonsobsd committed:
Update Makefile

Use textproc/opensearch like MASTERDIR

alonsobsd committed:
Delete textproc/opensearch210/files/opensearch.in

Delete obsolete file

alonsobsd committed:
Update Makefile

Use textproc/opensearch like MASTEDIR

alonsobsd committed:
Delete textproc/opensearch210/files/patch-config_jvm.options

Delete obsolete file

alonsobsd committed:
Delete textproc/opensearch210/files/patch-config_opensearch.yml

Delete obsolete file

alonsobsd committed:
Delete textproc/opensearch210/files/pkg-message.in

Delete obsolete file

alonsobsd committed:
Delete textproc/opensearch-dashboards210/pkg-descr

Delete obsolete file

alonsobsd committed:
Delete textproc/opensearch210/pkg-descr

Delete obsolete file

alonsobsd commented:
Hello, I did some modifications to opensearch*-210 ports. Please review those changes

smortex commented on textproc/opensearch-dashboards210/files/pkg-deinstall.in:
```diff
@@ -0,0 +1,20 @@
+#!/bin/sh
+
+case "$2" in
+POST-DEINSTALL)
+	if [ -d %%WWWDIR%%/optimize ]; then
+		/usr/bin/find %%WWWDIR%%/optimize/ -delete
+	fi
+
+	cat <<EOMSG
+
+If %%PORTNAME%%%%PKGNAMESUFFIX%% is being deleted permanently, and you do not wish to keep any
+data that was in the cluster, then you may wish to delete the
+%%WWWDIR%% directory.  This can be done by with the command:
+
+	# rm -rf %%WWWDIR%%
+
+Please ignore this if %%PORTNAME%%%%PKGNAMESUFFIX%% is being upgraded
+EOMSG
+    ;;
+esac
```
Should we not also have this for textproc/opensearch?

alonsobsd replied:
I think this file is not being included into opensearch-dashboards port/package. I don't see any SUB_FILES entry into Makefile file

smortex commented on textproc/opensearch/Makefile:
```diff
@@ -14,7 +14,7 @@ BUILD_DEPENDS=	jna>0:devel/jna
 		jna>0:devel/jna
 
-CONFLICTS=	opensearch13
+CONFLICTS?=	opensearch13 opensearch210
```
:+1: Yes, I forgot to update the CONFLICTS!

smortex commented on textproc/opensearch210/Makefile:
```diff
@@ -0,0 +1,13 @@
+
+MASTERDIR=      ${.CURDIR}/../../textproc/opensearch
+
+.include "${MASTERDIR}/Makefile"
```
Good idea I think

smortex reviewed (commented):
That looks good.  I think the pkg-deinstall.in was not intended for 2.10 only?

**Edit**: I can't approve this PR because I opened it :upside_down_face:

alonsobsd commented:
what about remove CHROMIUM from opensearch-dashboards OPTIONS?  Really it works from any browser based on chrome. I think the best is put a note into a pkg-message file or just remove the OPTION

alonsobsd committed:
Update Makefile

- Add pkg-deinstall to SUB_FILES
- Define SUB_LIST
- Remove CHOMIUM option

alonsobsd committed:
Delete textproc/opensearch-dashboards210/files/pkg-deinstall.in

Remove obsolete file

smortex commented:
> what about remove CHROMIUM from opensearch-dashboards OPTIONS? Really it works from any browser based on chrome. I think the best is put a note into a pkg-message file or just remove the OPTION

The option is not enabled by default, so no objection to tweak this. I personally don't use this feature :shrug:

smortex commented:
FYI, 2.11.1 is at the corner.  I guess we can wait for it before updating?

https://github.com/opensearch-project/opensearch-build/issues/4161

alonsobsd commented:
> FYI, 2.11.1 is at the corner. I guess we can wait for it before updating?
> 
> [opensearch-project/opensearch-build#4161](https://github.com/opensearch-project/opensearch-build/issues/4161)

I agree. I read it breaks logstash too if I'm not mistaken. Maybe it changes on 2.11.1

smortex commented:
> I agree. I read it breaks logstash too if I'm not mistaken. Maybe it changes on 2.11.1

If I understood correctly, 2.11.1 is intended to fix the regression that break logstash when compression is used in 2.10.0.  Not a user of logstash, so I am not affected by this.  I guess a few more fixes will be part of the new release.

hackacad committed:
update to 2.11.1

alonsobsd commented:
Hello, I  have committed opensearch* update and I have added 2.10.0 version too. I did some minor modifications to opensearch ports. Now it is more opensearch friendly ( right permissions to directories and config files) 

https://cgit.freebsd.org/ports/commit/?id=c7f140c561d4ab126286d67be6b8377a35d82f04
https://cgit.freebsd.org/ports/commit/?id=1132113d67323e1489cf89af10072cfae7dc36e5
